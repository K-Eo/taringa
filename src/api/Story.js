import APIBase from './APIBase'

class Story extends APIBase {
  byId(id) {
    return super.execute(`/story/${id}`)
  }

  comments(id, after = '') {
    return super.execute(`/story/${id}/comments`, { after })
  }
}

export default Story
