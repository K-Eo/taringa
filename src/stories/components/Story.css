.Story {
  border-radius: 0;
}

.Story ~ .Story {
  border-top: none;
}

.Story:first-child {
  border-top-left-radius: 0.25rem;
  border-top-right-radius: 0.25rem;
}

.Story:last-child {
  border-bottom-left-radius: 0.25rem;
  border-bottom-right-radius: 0.25rem;
}
