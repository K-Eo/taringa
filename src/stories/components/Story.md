Shout

```js
const shout = require('./data').StoryShout;
shout.summary.images.amount = 0;
<Story {...shout}/>
```

Shout with image thumbnail

```js
const shout = require('./data').StoryShout;
shout.summary.images.amount = 1;
<Story {...shout}/>
```

Shout with image thumbnail and preview

```js
const shout = require('./data').StoryShout;
shout.summary.images.amount = 1;
<Story {...shout} previewIsOpen/>
```
