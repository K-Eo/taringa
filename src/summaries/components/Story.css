.Story-preview {
  background-color: rgba(0, 0, 0, 0.05);
  margin-left: calc(-0.5rem - 1px);
  margin-right: calc(-0.5rem - 1px);
  margin-top: 0.5rem;
  padding: 0.5rem;
  text-align: center;
}
