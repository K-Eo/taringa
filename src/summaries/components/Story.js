import React from 'react'

import history from '../../history'

const Story = ({ slug, children }) => {
  const handleOnClick = () => {
    history.push(`/story/${slug}`)
  }

  return (
    <div
      className="list-group-item list-group-item-action p-2 Summary"
      onClick={handleOnClick}
    >
      {children}
    </div>
  )
}

export default Story
