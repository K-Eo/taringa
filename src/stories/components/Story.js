import React, { Component } from 'react'
import { connect } from 'react-redux'

import './Story.css'

import { UserCard } from '../../users/components/UserCard'
import StoryContent from './StoryContent'
import Comments from '../../comments/components/CommentsContainer'
import * as actions from '../actions'
import { slugToId } from '../../utils/slug'
import { storySelector, storyStateSelector } from '../selectors'

export class Story extends Component {
  componentDidMount() {
    this.props.fetchStoryWithComments()
  }

  render() {
    const { story, storyId, storyStatus: { status } } = this.props

    return (
      <div className="row">
        <div className="col-8">
          <StoryContent {...story} status={status} />
          <Comments storyId={storyId} />
        </div>

        <div className="col-4">
          <UserCard user={story.owner} control={{ status }} />
        </div>
      </div>
    )
  }
}

Story.defaultProps = {
  match: {
    params: {
      slug: '',
    },
  },
  storyStatus: {
    error: '',
    status: 'fetching',
  },
  fetchStoryWithComments: () => {},
  story: {},
}

const getStoryId = props => slugToId(props.match.params.slug)

const mapStateToProps = (state, props) => {
  const storyId = getStoryId(props)

  return {
    story: storySelector(state, storyId),
    storyId: storyId,
    storyStatus: storyStateSelector(state, storyId),
  }
}

const mapDispatchToProps = (dispatch, props) => {
  const storyId = getStoryId(props)

  return {
    fetchStoryWithComments: () => dispatch(actions.fetchWithComments(storyId)),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Story)
