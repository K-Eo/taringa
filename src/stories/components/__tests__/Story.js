import React from 'react'

import Story from '../Story'

describe('Story', () => {
  it('exists', () => {
    expect(Story).toBeDefined()
  })

  it('renders', () => {
    const wrapper = shallow(<Story />)

    expect(wrapper).toMatchSnapshot()
  })
})
