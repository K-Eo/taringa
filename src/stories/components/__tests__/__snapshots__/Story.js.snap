// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story renders 1`] = `
<div
  className="row"
>
  <div
    className="col-12 col-lg-8 mb-4 mb-lg-0 order-2 order-lg-1"
  >
    <Connect(lifecycle(Branch))
      storyId="id"
    />
    <div
      className="card my-4"
    >
      <div
        className="card-body"
      >
        <Connect(InfiniteScroll)
          storyId="id"
        />
      </div>
    </div>
  </div>
  <div
    className="col-12 col-lg-4 mb-4 mb-lg-0 order-1 order-lg-2"
  >
    <Connect(Connect(UserCardContainer))
      storyId="id"
    />
  </div>
</div>
`;
